activerecord/test/cases/transactions_test.rb

```diff
@@ -1660,23 +1660,23 @@ def test_no_automatic_savepoint_for_inner_transaction
 class TransactionUUIDTest < ActiveRecord::TestCase
   def test_the_uuid_is_lazily_computed
     Topic.transaction do
-      transaction = Topic.connection.current_transaction
+      transaction = Topic.current_transaction
       assert_nil transaction.instance_variable_get(:@uuid)
     end
   end
 
   def test_the_uuid_for_regular_transactions_is_generated_and_memoized
     Topic.transaction do
-      transaction = Topic.connection.current_transaction
+      transaction = Topic.current_transaction
       uuid = transaction.uuid
       assert_match(/\A[[:xdigit:]]{8}-(?:[[:xdigit:]]{4}-){3}[[:xdigit:]]{12}\z/, uuid)
       assert_equal uuid, transaction.uuid
     end
   end
 
-  def test_the_uuid_for_null_transactions_is_the_nil_uuid
-    null_transaction = ActiveRecord::ConnectionAdapters::TransactionManager::NULL_TRANSACTION
-    assert_equal Digest::UUID.nil_uuid, null_transaction.uuid
+  def test_the_uuid_for_null_transactions_is_nil
+    null_transaction = ActiveRecord::Transaction::NULL_TRANSACTION
+    assert_nil null_transaction.uuid
```

Comment:
Make sense, let me think about this detail. At the moment I was pursuing the idea that the null object is meant to be a transparent substitute. If we say: the transaction has a UUID unless it is a null object, then we introduce a difference which I am in favor of, if you followed the documentation discussion, but unsure it is in line with the interface Jean has in mind.

Reply:
It is also worth noting that `transaction.active_record` is not triggered for null objects, and that `sql.active_record` does not receive null objects (though subscribers would get one if asking for `current_transaction` instead of using the payload).